text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,333 @@
+
+By removing this set of exports, we have an opportunity to shrink some apps (as some APIs are not used), improving the load performance of ember apps -- and we removing all of these gives us a chance to have a better grasp of what we can get rid of permananently.
+
+Many of these APIs already have alternatives, and those will be called out explicitly in the _Transition Path_ below.
+
+## Transition Path
+
+This list is semi-exhaustive, in that it covers _every_ export from 'ember', but may not exhaustivily provide alternatives.
+
+### New Module Needed
+
+APIs for wiring up a test framework (e.g. QUnit, _etc_)
+- `Ember.Test`
+- `Ember.Test.Adapter`
+- `Ember.Test.QUnitAdapter`
+- `Ember.setupForTesting`
+
+These will need to be moved to a module such as `@ember/testing`.
+
+### A way to communicate with the ember-inspector
+
+The inspector will be hit especially hard by the removal of these APIs.
+
+- `Ember.meta`
+- `Ember.VERSION`
+- `Ember._captureRenderTree` 
+- `Ember.instrument`
+- `Ember.subscribe`
+- `Ember.Instrumentation.instrument`
+- `Ember.Instrumentation.subscribe`
+- `Ember.Instrumentation.unsubscribe`
+- `Ember.Instrumentation.reset`
+- `Ember.ViewUtils`
+- `Ember.ViewUtils.getChildViews`
+- `Ember.ViewUtils.getElementView`
+- `Ember.ViewUtils.getRootViews`
+- `Ember.ViewUtils.getViewBounds`
+- `Ember.ViewUtils.getViewBoundingClientRect`
+- `Ember.ViewUtils.getViewClientRects`
+- `Ember.ViewUtils.getViewElement`
+- `Ember.ViewUtils.isSimpleClick`
+- `Ember.ViewUtils.isSerializationFirstNode`
+
+
+Perhaps we can have folks add this to their apps:
+```js
+import { macroCondition, isDevelopingApp, importSync } from '@embroider/macros';
+
+if (macroCondition(isDevelopingApp())) {
+  // maybe this is side-effecting and installs 
+  // some functions on `globalThis` that the inspector could call
+  importSync('@ember/inspector-support');
+}
+```
+
+### No replacements.
+
+Applies to both the value and type exports (if applicable).
+
+- `Ember._getPath`
+- `Ember.isNamespace`
+- `Ember.toString`
+- `Ember.Container`
+- `Ember.Registry`
+
+Internal decorator utils
+- `Ember._descriptor`
+- `Ember._setClassicDecorator`
+
+Reactivity
+- `Ember.beginPropertyChanges`
+- `Ember.changeProperties`
+- `Ember.endPropertyChanges`
+
+Observable 
+- `Ember.hasListeners`
+
+Mixins
+- `Ember._ContainerProxyMixin`
+- `Ember._ProxyMixin`
+- `Ember._RegistryProxyMixin`
+- `Ember.ActionHandler`
+- `Ember.Comparable`
+
+Utility
+- `Ember.libraries` - 
+   App authors could choose to use any webpack or other build plugin that collections this information, such as [webpack-node-modules-list](https://github.com/ubilabs/webpack-node-modules-list). This additionally means that V1 libraries that pushed themselves into `Ember.libraries` no longer need to worry about interacting with this or any similar API. 
```

Comment:
It is also used in ember inspector to display registered libraries and their versions. 
I think it was nice. I'm not sure if inspector can use the generated meta data. But if it can it would also be nice.

Reply:
maybe there is an unpulgin that we can suggest folks use in both webpack and vite-using apps that can push data in to the inspector.

or maybe, since the logic is tiny anyway, we make some ember-inspector plugins that emit a known virtual module, imported by the app boot virtual module (or directly in app.js, idk, theory-crafting here).

and we have a family of plugins:
```js
'ember-inspector-plugins/libraries',
'ember-inspector-plugins/version-info',
```

Reply:
https://github.com/emberjs/rfcs/pull/1003/files#r1491746149